rest/admin-api/src/main/java/org/keycloak/admin/api/AdminRootV2.java

```diff
@@ -0,0 +1,51 @@
+package org.keycloak.admin.api;
+
+import jakarta.ws.rs.OPTIONS;
+import jakarta.ws.rs.Path;
+import jakarta.ws.rs.core.Context;
+import jakarta.ws.rs.core.Response;
+import jakarta.ws.rs.ext.Provider;
+import org.eclipse.microprofile.openapi.annotations.Operation;
+import org.keycloak.common.Profile;
+import org.keycloak.models.KeycloakSession;
+import org.keycloak.services.ErrorResponse;
+import org.keycloak.services.resources.admin.AdminCorsPreflightService;
+
+@Provider
+@Path("admin/api")
+public class AdminRootV2 {
+
+    @Context
+    protected KeycloakSession session;
+
+    @Path("")
+    public AdminApi latestAdminApi() {
+        checkApiEnabled();
+        // we could return the latest Admin API if no version is specified
+        return session.getProvider(AdminApi.class);
+    }
+
+    @Path("v2")
+    public AdminApi adminApi() {
```

Comment:
We explicitly don't want versioning on the Admin API as whole; but rather on each distinct API.

Reply:
Created follow-up issue:
- https://github.com/keycloak/keycloak/issues/43571

But we need to discuss it within our SIG.